conans/model/options.py

```diff
@@ -390,7 +390,7 @@ def get_upstream_options(self, down_options, own_ref, is_consumer):
         # compute now the necessary to propagate all down - self + self deps
         upstream_options = Options()
         for pattern, options in down_options._deps_package_options.items():
-            if ref_matches(own_ref, pattern, is_consumer=is_consumer):
+            if ref_matches(own_ref, pattern, is_consumer=is_consumer) and "*" not in pattern:
```

Comment:
This gets expressions such as `pkg/*:shared=True` to the upstream options, which after thinking a bit about it, I _think_ does not really matter, right?

Reply:
I thought about it, it really doesn't matter, I was thinking about cases of boostrapping, when you can require yourself for building, but in practice this doesn't happen because that is only possible in ``tool_requires``, and those do not propagate the options (I have added more tests).

I'd also like to keep the propagated options clean if possible (they will be polluting build-orders and other stuff), so I have restricted it to the "name" part of the pattern.